eng/pipelines/templates/jobs/smoke.tests.yml

```diff
@@ -185,7 +185,22 @@ jobs:
           env:
             REQUIREMENTS: $(requirements)
 
-        - pwsh: pip install -r "$env:REQUIREMENTS" --no-deps --upgrade
+        - pwsh: |
+            $timeout = New-TimeSpan -Seconds 600
```

Comment:
@benbp @hallipr it is starting to look like having some sort of retry helper function might be useful in our common helpers.